### PR TITLE
Fix panic when invoked on crate missing metadata (#377)

### PR DESCRIPTION
**Pull request by gkelly**
Also add a test that ensures the error is reported and a panic doesn't
occur.

### CODE REVIEW

**Comment by gkelly** [comment]
With this patch instead of panicking the result is now:

```
Error: Raze failed with cause: "Exactly one package must contain primary raze settings: []"
```

File: impl/src/settings.rs [hunk 1]
```diff
@@ -626,9 +626,9 @@ fn parse_raze_settings_any_package(metadata: &Metadata) -> Result<RazeSettings>
   }
 
   // There should only be one package with raze
-  if settings_packages.len() > 1 {
+  if settings_packages.len() != 1 {
```
Comment (dfreese) [inline impl/src/settings.rs:629]: It's probably worth specifying the zero case separately.  Perhaps something like:
>No raze settings were specified in the Cargo.toml file, see README.md for details on expected fields

would be more helpful?

also, not your doing, but I'm noticing that
```rust
return Err(anyhow!("error message"));
```
is used.  Anyhow provides anyhow::bail
```rust
bail!("error message");
```
that does the same thing, more concisely.

**Review by dfreese — COMMENTED** [review]
Thanks for the help providing clearer errors.

File: impl/src/settings.rs (hunk 1 above)
Comment (gkelly) [reply impl/src/settings.rs]: > It's probably worth specifying the zero case separately. Perhaps something like:
> 
> > No raze settings were specified in the Cargo.toml file, see README.md for details on expected fields
> 
> would be more helpful?

Done. I do worry that it should perhaps be a URL, as if someone is just running cargo-raze as part of CI, etc, that it won't be clear which `README.md` is intended in context.

> also, not your doing, but I'm noticing that
> 
> ```rust
> return Err(anyhow!("error message"));
> ```
> 
> is used. Anyhow provides anyhow::bail
> 
> ```rust
> bail!("error message");
> ```
> 
> that does the same thing, more concisely.

Done.

**Comment by dfreese** [comment]
great, thanks!

**Comment by gkelly** [comment]
> great, thanks!

I appreciate the reviews and merge, thanks!

**Pull request merged**